lib/crystalball/source_diff.rb

```diff
@@ -28,7 +28,18 @@ def empty?
 
     # TODO: Include untracked to changeset
     def changeset
-      @changeset ||= git_diff.map { |file_diff| FileDiff.new(file_diff) }
+      @changeset ||= git_diff.map do |file_diff|
+        diff = FileDiff.new(file_diff)
+        # FIXME: ruby-git marks moved files as modified. It works for lib files, but not for specs
+        # This code splits diff of moved file to two separate diffs with removed and added file
+        if diff.moved?
+          added_file_diff = FileDiff.new(file_diff, override_type: 'new', override_relative_path: diff.new_relative_path)
+          removed_file_diff = FileDiff.new(file_diff, override_type: 'deleted')
+          [removed_file_diff, added_file_diff]
+        else
+          diff
+        end
+      end.flatten
```

Comment:
this looks weird. can you elaborate more on it?

Reply:
Git will mark a file as `modified` when you rename it. It works well for source files. But not for spec files. It will try to run a file with the old name and completely ignores the new file.
If spec file was renamed, it means that previous filename should be removed from prediction map and the new one should be added.
I do it manually. If patch indicates that file was moved, I manually create a diff with deleted file and with the added one

Reply:
done